### PR TITLE
OverlayCard component

### PR DESCRIPTION
**Pull request by ann-schulte-flatironschool**
**Additions:**

- OverlayCard Component
Screen Shot 2021-03-03 at 3 23 43 PM

### CODE REVIEW

File: src/components/OverlayCard/OverlayCard.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    background-color: ${props => Color[props.backgroundStyle]};
+    color: ${Color.black};
+    outline: 0;
+  }
+`
+
+export const OverlayCard = ({ cta, title, backgroundStyle, onClick }: OverlayCardProps): JSX.Element => (
+  <Container>
+    <StyledGrid backgroundStyle={backgroundStyle}>
+      {backgroundStyle === 'red' && (
```
Comment (tmilewski) [inline src/components/OverlayCard/OverlayCard.tsx:53]: Any reason we're using red for pink, turq for blue? We can roll these up into one call if we consolidate the naming?

File: src/components/OverlayCard/OverlayCard.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      )}
+    </StyledGrid>
+    <Grid item xs={12}>
+      {React.isValidElement(cta) ? (
+        cta
+      ) : (
+        <StyledButton onClick={onClick} backgroundStyle={backgroundStyle}>
+          {cta as string}
+        </StyledButton>
+      )}{' '}
```
Comment (tmilewski) [inline src/components/OverlayCard/OverlayCard.tsx:81]: Why the added space here?

File: src/components/Icon/Icon.d.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React, { SVGProps } from 'react'
 
 export type IcoProps = {
   color?: string
-  children?: never
+  children?: React.ReactNode
```
Comment (tmilewski) [inline src/components/Icon/Icon.d.ts:5]: I'd suggest using the SVG as a background or having the text as an absolute element on top.

All of this said, I'd also suggest not including the assets here and making them available for use elsewhere. Then place them in via a prop.

By putting them here, they get bundled with everything and slow things down (see: loading gif).

Using a prop, the consumer can define what's used and how they're loaded (bundled/external assets/etc.)

File: src/components/OverlayCard/OverlayCard.tsx (hunk 1 above)
Comment (tmilewski) [reply src/components/OverlayCard/OverlayCard.tsx:53]: That said, see my original comment about using props.

**Review by tmilewski — CHANGES_REQUESTED** [review]
Left a few comments. Is there a better name for this? I was very confused as to what it was until I dug in.

File: src/components/OverlayCard/OverlayCard.tsx (hunk 2 above)
Comment (ann-schulte-flatironschool) [reply src/components/OverlayCard/OverlayCard.tsx:81]: accident

**Review by tmilewski — APPROVED** [review]

**Pull request merged**